lib/ansiblelint/runner.py

```diff
@@ -101,6 +103,27 @@ def run(self) -> List:
                 "Examining %s of type %s",
                 ansiblelint.file_utils.normpath(file['path']),
                 file['type'])
+
+            # we should bother checking playbooks only if they pass Ansible syntax-check
+            if file['type'] == 'playbook':
+                result = subprocess.run(
+                    ['ansible-playbook', '--syntax-check', file['path']],
+                    stdin=subprocess.PIPE,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    shell=False,  # needed when command is a list
+                    universal_newlines=True,
+                    check=False
+                )
+                if result.returncode != 0:
+                    matches.append(MatchError(
+                        message="Ansible syntax-check failed, maybe file is not a playbook.",
+                        details="\n".join([result.stdout, result.stderr]),
+                        filename=file['path'],
+                        rule=AnsibleSyntaxCheckRule
+                    ))
+                    continue
```

Comment:
the first step would be to do this
```suggestion
                try:
                    ansible_playbook_check_output = subprocess.check_output(
                        ['ansible-playbook', '--syntax-check', file['path']],
                        stderr=subprocess.STDOUT,
                        shell=False,  # needed when command is a list
                        universal_newlines=True,
                    )
                except subprocess.CalledProcessError:
                    matches.append(MatchError(
                        message="Ansible syntax-check failed, maybe file is not a playbook.",
                        details=ansible_playbook_check_output,
                        filename=file['path'],
                        rule=AnsibleSyntaxCheckRule,
                    ))
                    continue
```